src/librustc_mir/interpret/operand.rs

```diff
@@ -271,16 +271,18 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M>
         if mplace.layout.is_zst() {
             // Not all ZSTs have a layout we would handle below, so just short-circuit them
             // all here.
-            self.memory.check_align(ptr, ptr_align)?;
+            self.memory.check_align(ptr, ptr_align.min(mplace.layout.align))?;
```

Comment:
Why this `min`? Seems like if the place is sure that the alignment is higher than the layout, then we should just go with that.